internal/pkg/application/application.go

```diff
@@ -1,48 +1,199 @@
+}
+
+func New(baseUrl, accountID, accountKey string, log zerolog.Logger, cb client.ContextBrokerClient) IntegrationAcoem {
+	return &integrationAcoem{
+		baseUrl:    baseUrl,
+		accountID:  accountID,
+		accountKey: accountKey,
+		log:        log,
+		cb:         cb,
+	}
+}
+
+func (i *integrationAcoem) CreateAirQualityObserved(ctx context.Context) error {
+	headers := map[string][]string{"Content-Type": {"application/ld+json"}}
+
+	var err error
+	if i.accountID == "" || i.accountKey == "" {
+		log.Error().Err(err).Msg("account id and account key must not be empty")
 	}
 
-	parsedInterval, err := strconv.Atoi(interval)
+	stations, err := i.getStations()
 	if err != nil {
-		log.Err(err).Msg("failed to parse string interval to integer")
+		log.Error().Err(err).Msg("failed to retrieve stations")
 		return err
 	}
 
-	for {
-		stations, err := GetStations(baseUrl, accountID, accountKey)
+	for _, stn := range stations {
+
+		sensors, err := i.getSensorData(stn)
 		if err != nil {
-			log.Err(err).Msg("failed to retrieve stations")
+			log.Error().Err(err).Msg("failed to retrieve sensor data")
 			return err
 		}
 
-		for _, stn := range stations {
-			result, err := GetSensorData(baseUrl, accountID, accountKey, stn)
-			if err != nil {
-				log.Err(err).Msg("failed to retrieve sensor data")
-				return err
-			}
+		id := fiware.AirQualityObservedIDPrefix + strconv.Itoa(stn.UniqueId)
+
+		decorators := []entities.EntityDecoratorFunc{}
 
-			stnBytes, err := json.MarshalIndent(result, "", "  ")
+		for _, sensor := range sensors {
+
+			observed, err := time.Parse(time.RFC3339, sensor.TBTimestamp)
 			if err != nil {
-				log.Err(err).Msg("failed to marshal data into structs")
-				return err
+				observed = time.Time{}
 			}
 
-			fmt.Printf("latest data from %s station: %s", stn.StationName, string(stnBytes))
+			decorators = append(decorators,
+				Location(sensor.Latitude, sensor.Longitude),
+				DateTimeIfNotZero(properties.DateObserved, observed),
+			)
+
+			sensorReadings := createFragmentsFromSensorData(sensor.Channels)
+
+			decorators = append(decorators, sensorReadings...)
+		}
+
+		entity, err := entities.New(id, fiware.AirQualityObservedTypeName, decorators...)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to create new entity")
+		}
+
+		jsonEnt, _ := json.MarshalIndent(entity, " ", "  ")
+		fmt.Printf("entity: %s", jsonEnt)
+
+		_, err = i.cb.CreateEntity(ctx, entity, headers)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to post entity to context broker")
 		}
 
-		time.Sleep(time.Duration(parsedInterval) * time.Second)
 	}
+
+	return nil
+}
+
+var unitCodes map[string]string = map[string]string{
+	"Micrograms Per Cubic Meter": "GQ",
+	"Volts":                      "VLT",
+	"Celsius":                    "CEL",
+	"Percent":                    "P1",
+	"Hectopascals":               "A97",
+	"Parts Per Billion":          "61",
+	"Pressure (mbar)":            "MBR",
+	"Particles per cm3":          "Particles per cm3",
```

Comment:
You can't pass this "propblemapa" to the broker. Figure this unit code out or omit the value.